### ISSUE TITLE
Remove POP3 support

### ISSUE THREAD

**Issue by cketti** [k9mail/k-9#4974]
Remove support for the POP3 protocol.

K-9 Mail never supported POP3's "download then delete from server" model. The app tries to treat POP3 accounts similar to IMAP where we download copies of messages but treat the server as single source of truth. Some settings exist to weaken that model, but the goal of #4970 is to get rid of these, too.

The hardest part about the removal is a deprecation strategy. We'll have to inform users that support for POP3 is going away before we actually remove it. They should have plenty of time to prepare moving to another protocol or another email client. I propose we show a non-dismissable info message above the message list. Clicking it could lead to a blog post with more information on the topic.

I don't want us to automatically delete user data. So once POP3 support is removed I propose we convert remaining POP3 accounts to read-only accounts. Users will still be able to view messages. They won't be able to copy, move, delete, or compose new messages from that account.

*This is part of the effort to [remove support for local messages](https://github.com/k9mail/k-9/issues/4970).*

**Comment by bitshftr** [comment]
Noooo....don't remove POP3 support.  I use POP3 for my work email on my personal phone, but only when I'm traveling.  I don't want to use IMAP or sync all of my email onto my phone.  I only want to receive on my phone those messages that have come in since I left the office.  I do not want those messages removed from the server, or even marked as "read" on the server.

**Comment by cketti** [comment]
With IMAP you don't have to sync all of your emails to your phone. K-9 Mail doesn't support only receiving messages "that have come in since [you] left the office", whether you use POP3 or IMAP. You can configure the app to not mark messages as read when they are opened.

**Comment by bitshftr** [comment]
` K-9 Mail doesn't support only receiving messages "that have come in since [you] left the office"`

I'm not asking for K-9 to know what came in since I left the office.  I use POP3 in Thunderbird on my office desktop computer, and I have it set to delete the messages from the server.  So when I leave the office, there's nothing left on the server's "inbox".  Then as emails come in while I'm away, I can manually invoke pulling emails on my phone, and I'll get emails that were sent to me after I left the office (after I last checked email from my desktop computer).

**Comment by cketti** [comment]
I don't want this to be a "your idea about POP3 is not entirely correct" support ticket. Please use the [forum](https://forum.k9mail.app/) if you wish to discuss this further. I'm happy to go into details there.

**Comment by cyberbeat** [comment]
Sorry, but I have a similar use case as the previous commenter.
a) I don't want to have my emails be stored on any server longer then required (security, privacy,..), but only locally
b) I have a desktop pc, which deletes the mails automatically after download
c) I have a phone, with which I can read mails from the server, but does not delete them from the server

Maybe this scenario would be possible with IMAP in theory, too, but no email-client I know does support this for IMAP.

Second: you write above "They should have plenty of time to prepare moving to another protocol or another email client". To which email client? Most newer clients also don't support pop3 anymore.

I do not understand, why you mark such use cases as off-topic.

**Comment by cketti** [comment]
Most people use POP3 because they want to delete messages from the server after downloading them. K-9 Mail does not support this at all, whether you use POP3 or IMAP doesn't matter. Given that fact, there's really no reason to use POP3 *with K-9 Mail*. 

Most people don't actually want to delete messages from the server after downloading them to their mobile device. They do that after downloading messages to their desktop. Again, there's no good reason to use POP3 *on mobile*. IMAP is much better at downloading messages and leaving them on the server than POP3 is.

There's no need to only use a single protocol to access emails. It's totally fine to use POP3 on desktop and IMAP on mobile.

**Comment by mavoga** [comment]
Please don't do that. ***There are* reasons for using POP3**: whether IMAP4 is available or not does not always depend on users, who are often forced to use one, the other or even none (EWS only) by admins, especially in the times of Office365 when enabling a protocol is just a click of the mouse.

POP3 support is there and works well (it only lacks OAuth2 support, please see #6216), removing it would be a serious loss and would force users to install a second mail client just for POP3 only mailboxes.

Please avoid adhering to a "developer decides all, users must adapt" model.

**Comment by ipfreer** [comment]
to add a voice to the keep POP3 group. please consider yourself as a single voice and not necessarily speaking for humanity, that there may be many people like myself for near 3 years now that have no desktop/laptop access only their phone, and that not every email service out there supports IMAP or is Trustworthy enough to keep all mail on server or provides enough storage for leaving mail on server, or doesn't back up server reliably or encourages users to do anything but get mail off their servers ASAP as many Privacy and Security focused servers do and in fact often give very small amounts of storage for both free and paid accounts (eg. I have 2 free with 1GB storage and 2 paid by another provider with 500MB storage).

Dev, please consider you are not alone anymore, I know this is your baby, it was my first cell phone email client in 2009 iirc, but your days as sole voice and decision maker for this project are in the past now and you have not only a userbase to consider you now have a significantly much larger Thunderbird family that I can only imagine, like me, wants their phone mail client (k9 mail) to function and probably appear like their desktop client does. we're not asking you to add feature, nor ability for add-ons, extensions, skins and the incredible tweakableness of Thunderbird desktop, we're asking you to Not remove a very useful, arguably necessary, feature already a part of k9 for over a decade.

in the end, and I imagine you kept final decision on k9 rights in the transfer to Thunderbird family, but surely you must see that not everyone has access to other services, not everyone has trustworthy servers, nor ones that provide enough online storage space, not everyone can be so easy to make other choices on apps usable (especially now you work very well with PGP encryption).

You might consider it better to get more democratic on such decisions polling in Thunderbird forums making it available and open for comments for period of 30-60 days with any Thunderbird announcement emails, and front page news on site homepage pointing to the do we want/need secure POP3 in K9?

please take no offense from my pleas, I mean no harm nor insult. I know coding isn't simple and it rarely puts food on the table for even coders in the largest and most successful FOSS projects. I wish my brain worked well enough to even learn coding so I could contribute. and while I've donated to k9 one time, many years ago, I admit it was only once and that $20 was nothing to the work performed nor my use case for k9. I also admit I had to leave k9, maybe for good I'm not sure, but at least until I got a home computer hoping for NLT this summer. I've been locked to using phone only for near 3 yrs now and those email services I mentioned above are now filling or full. I cannot, for legal, medical and other personal reasons remove any mail from server until I have a verified local backup. I see learned a very hard lesson about k9 mail last year when my old phone died and it had to do with losses of emails. in desperation I tried every app and service to no avail because like many I am also not on a rooted phone. I recently found responsive developer, who seemed much like you do, to consider POP3 as less feature rich or desirable than IMAP for every modern smartphone user on earth. I may have swayed his opinion on both the true benefits of POP3 and to find a way to allow non root users to backup their email clients, both settings and any local stores messages, attachments, etc. He's working on fixing and further developing POP3 capabilities in app, and looks to be testing some backup function for emails in recent releases. I've contributed a good sum from fixed income, and will continue to because he didn't dig in and say no to very valid use case and reasonable realistic features that can only grow in popularity as many countries cannot afford entry into Internet world with desktops, laptops and all manner of cloud services and such. but maybe they can afford a second hand cell phone and like me they either need or want to keep emails in local archived storage. There is also, I imagine like here in the US, many around the world are on the move, all the time, and even those that work remote, many seem to be migrating to simplify life, declutter, reduce footprint and unneeded expenses, like laptop and desktop can become with modern smartphones being so incredibly powerful and fast and secure!

I believe you come from an older generation, like I do. only difference, I think, may be for past 25 yrs I was always using POP3 any time I could because I didn't want to leave mail on servers for both security and Privacy reasons. while at work for two decades in the Army I had to use IMAP (and exchange), or Webmail. I disliked those choices but certainly saw their use cases and benefits. and really I find most times a service is assure to keep mail on server it's either corporation/work mail or it's free service like gmail that made billion$$ by having that access to email at rest, just keep growing the free storage, forever. (fwiw, I was a gmail beta test account, it's still my primary gmail too). Email is just convenient snailmail, post cards at that. POP3 is like an honest Post Office and user is renting a PO box. my PO box is full and I am prohibited from taking it all home and securing it in local firesafe box.

sorry this is so long, and maybe errors included, screen is small edit window is tiny and swiping finger is huge.

bottom line, please do not remove this functionally without at least being open for comment by entire community and family you are now a part of.

**Comment by mwat56** [comment]
**Keep POP3!**
For me and my clients it's _essential_. There's nor reason to remove a valued part of K9.

**Comment by viktak** [comment]
Agreed - do NOT remove pop3 support...

**Comment by mwat56** [comment]
Keep POP3, please. For me POP3 support is essential since I want to keep my emails on a server that's accessable from the internet for the smallest time possible. I download the mails automatically from my phone, my tablet, and my PC and then have them automatically removed. Without POP3 support I would have to replace Thunderbird by something else.
BTW: Why would one want to remove a well established mail protocol at all?

**Comment by ghost** [comment]
Who's using POP3 in 2026!?

I do, because IMAP failed to delete emails after fetching several times and proven untrustworthy from privacy perspective.

POP3 is old, but old isn't bad. It's perfection.

May POP3 live 100 years more!
